### PR TITLE
Feature - Newsroom Subscriptions API

### PR DESCRIPTION
**Pull request by M4rk07**
- Deprecate `NewsroomSubscriptions` SDK.
- Implement `Subscriptions` SDK.

### CODE REVIEW

File: src/Sdk/Subscriptions/types.ts
Comment (kamilmielnik) [inline src/Sdk/Subscriptions/types.ts:54]:
```suggestion
        address?: Partial<Contact['address']>;
```

File: src/Sdk/Subscriptions/types.ts
Comment (kamilmielnik) [inline src/Sdk/Subscriptions/types.ts:14]: This and other similar comments can be expressed with TS:

```tsx

interface Contact {
    contact_type: ContactType;
    first_name: string;
    company_name: string;
}

interface NewsroomSubscribeRequest<Type extends Contact['contact_type']> {
    /* ... */
    contact?: (Type extends 'person' ? Person : Organisation) & {
        // common props go here
        languages?: string[];
    };
}

interface Person {
    first_name: Contact['first_name'];
    /* other person-specific attributes go here */
}

interface Organisation {
    company_name: Contact['company_name'];
    /* other organisation-specific attributes go here */
}

// examples:
const x: NewsroomSubscribeRequest<ContactType.PERSON> = {
    contact: {
        languages: [], // ok
        company_name: 'asd', // error
        first_name: 'asd', // ok
    },
};

const y: NewsroomSubscribeRequest<ContactType.ORGANISATION> = {
    contact: {
        languages: [], // ok
        first_name: 'asd', // error
        company_name: 'asd', // ok
    },
};

```

Comment (M4rk07) [reply src/Sdk/Subscriptions/types.ts]: @kamilmielnik thank you for this tip/suggestion!

Please check the changes. 

I also added `contact_type` fields to `Person` and `Organisation` interfaces. Also check how I specified type of `payload` parameter of `subscribeToNewsroom` method.

**Review by e1himself — APPROVED** [review]
Looks good.

Let's also wait for at least one review from the javascript team :)

File: src/Sdk/Subscriptions/types.ts
Comment (kamilmielnik) [inline src/Sdk/Subscriptions/types.ts:9]: None of these attributes are required - is this correct?

Because previously comments in this file stated:

> Required only when contact_type === 'person'

File: src/Sdk/Subscriptions/Subscriptions.ts
Comment (kamilmielnik) [inline src/Sdk/Subscriptions/Subscriptions.ts:22]: > Also check how I specified type of payload parameter of subscribeToNewsroom method.

We can make it even better :)


```suggestion
    public async subscribeToNewsroom<Type extends ContactType>(
        newsroomId: NewsroomId,
        payload: NewsroomSubscribeRequest<Type>,
    ): Promise<EmailSubscription> {
```

File: src/Sdk/Subscriptions/types.ts
Comment (M4rk07) [reply src/Sdk/Subscriptions/types.ts:9]: Great catch! But it was a wrong doc before. 😬  These parameters are actually not required.

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Pull request merged**
